ui/app/src/matrix-store.ts

```diff
@@ -1341,6 +1352,34 @@ export class MatrixStore {
 
       const hcPort = import.meta.env.VITE_AGENT === "2" ? import.meta.env.VITE_HC_PORT_2 : import.meta.env.VITE_HC_PORT;
       const appletAppAgentWebsocket = await AppAgentWebsocket.connect(`ws://localhost:${hcPort}`, appInfo.installed_app_id);
+      
+      
+      const devhubHappReleaseHash =
+        this.releaseHashOfAppletInstance(appletInstanceId)!;
+      // ATTENTION: IT IS ASSUMED HERE THAT THE APPLET IS ALREADY IN THE MATRIX!!
+
+      let gui = this._appletGuis.get(devhubHappReleaseHash);
+      if (!gui) {
+        gui = await this.queryAppletGui(devhubHappReleaseHash);
+      }
+
+      // register applet config and widgets on applet join
```

Comment:
Are these Sensemaker `Dimension` widgets that Applets can provide? Just want to be specific about terminology to distinguish from `Resource` rendering widgets.

Reply:
good point. Yeah, these are *sensemaker dimension* widgets, will add that to the comment.